### PR TITLE
feat(crons): Move errors to issue platform

### PR DESCRIPTION
**Pull request by rjo100**
image

Initial moving of monitor errors to issue platform

### CODE REVIEW

File: src/sentry/monitors/models.py [hunk 1]
```diff
@@ -444,21 +452,46 @@ def mark_failed(self, last_checkin=None, reason=MonitorFailure.UNKNOWN):
...
+                "event_id": uuid.uuid4().hex,
+                "platform": "crons",
+                "project_id": self.monitor.project_id,
+                "received": timezone.now(),
+                "sdk": None,
+                "tags": {
+                    "monitor.id": str(self.monitor.guid),
+                    "monitor.slug": self.monitor.slug,
+                },
+                "timestamp": timezone.now(),
```
Comment (rjo100) [inline src/sentry/monitors/models.py:467]: can be problematic if we want to say last good check-in (and that was > 1 month ago)

File: src/sentry/monitors/models.py [hunk 2]
```diff
@@ -444,21 +452,46 @@ def mark_failed(self, last_checkin=None, reason=MonitorFailure.UNKNOWN):
...
             },
-            project=Project(id=self.monitor.project_id),
+            "issue_title": f"Monitor failure: {self.monitor.name} ({reason})",
+            "subtitle": f"{self.environment.name}",
+            "resource_id": None,
+            "evidence_data": {"Timestamp": last_checkin},
```
Comment (rjo100) [inline src/sentry/monitors/models.py:472]: this is a safe space to put `last_checkin`

File: src/sentry/monitors/models.py [hunk 3]
```diff
@@ -481,21 +489,46 @@ def mark_failed(self, last_checkin=None, reason=MonitorFailure.UNKNOWN):
...
+
+        producer.produce(
+            topic=topic,
+            key=None,
+            value=json.dumps(failure_occurrence, default=str),
         )
-        event_manager.normalize()
-        data = event_manager.get_data()
-        insert_data_to_database_legacy(data)
+        producer.flush()
```
Comment (barkbarkimashark) [inline src/sentry/monitors/models.py:531]: I think we can replace all this with: https://github.com/getsentry/sentry/blob/9d47aa4261013f4636408ac97ff2a50a13af7491/src/sentry/issues/producer.py#L28

File: src/sentry/monitors/models.py [hunk 4]
```diff
@@ -481,21 +489,46 @@ def mark_failed(self, last_checkin=None, reason=MonitorFailure.UNKNOWN):
...
+        producer = Producer(cluster_options)
+
+        producer.produce(
+            topic=topic,
+            key=None,
+            value=json.dumps(failure_occurrence, default=str),
         )
-        event_manager.normalize()
-        data = event_manager.get_data()
-        insert_data_to_database_legacy(data)
```
Comment (barkbarkimashark) [inline src/sentry/monitors/models.py:498]: Might make sense to feature flag this so you can switch between creating issue through issue platform versus the old way.

File: src/sentry/monitors/models.py [hunk 5]
```diff
@@ -481,21 +485,84 @@ def mark_failed(self, last_checkin=None, reason=MonitorFailure.UNKNOWN):
...
+                detection_time=current_timestamp,
+                level=level,
+            )
+
+            produce_occurrence_to_kafka(
+                occurrence,
+                {
+                    "environment": self.environment.name,
+                    "event_id": occurrence.event_id,
+                    "platform": "crons",
```
Comment (wedamija) [inline src/sentry/monitors/models.py:531]: I'm not sure if `crons` will work as a platform, it's usually specific to the framework that the event was fired from.

**Review by wedamija — COMMENTED** [review]
Generally lgtm, what's the migration plan here?

**Review by wedamija — COMMENTED** [review]
Generally lgtm, what's the migration plan here?

File: src/sentry/monitors/models.py (hunk 5 above)
Comment (wedamija) [reply src/sentry/monitors/models.py]: I'm not sure if empty works, but you could try it. Based on the docs you might want `other` https://develop.sentry.dev/sdk/event-payloads/

File: src/sentry/monitors/models.py [hunk 6]
```diff
@@ -481,21 +485,84 @@ def mark_failed(self, last_checkin=None, reason=MonitorFailure.UNKNOWN):
...
+
+            occurrence = IssueOccurrence(
+                id=uuid.uuid4().hex,
+                resource_id=None,
+                project_id=self.monitor.project_id,
+                event_id=uuid.uuid4().hex,
+                fingerprint=[f"monitor-{str(self.monitor.guid)}-{reason}"],
+                type=group_type,
+                issue_title=f"Monitor failure: {self.monitor.name} ({reason})",
+                subtitle="",
```
Comment (wedamija) [inline src/sentry/monitors/models.py:512]: I have a feeling we'll want a value here and in culprit. Writing up docs on how these are used now actually

**Review by wedamija — APPROVED** [review]

**Pull request merged**